examples/zynq-mpsoc/ad08-sequential.cpp

```diff
@@ -0,0 +1,244 @@
+  CYNQ_INFO("Creating memory");
+  std::shared_ptr<IMemory> buf_mem_mm_a =
+      mover->GetBuffer(size_a * sizeof(DataType), matmul->GetMemoryBank(0));
+  std::shared_ptr<IMemory> buf_mem_mm_b =
+      mover->GetBuffer(size_b * sizeof(DataType), matmul->GetMemoryBank(1));
+  std::shared_ptr<IMemory> buf_mem_mm_c =
+      mover->GetBuffer(size_c * sizeof(DataType), matmul->GetMemoryBank(2));
+  std::shared_ptr<IMemory> buf_mem_ew_a =
+      mover->GetBuffer(size_c * sizeof(DataType), elemwise->GetMemoryBank(0));
+  std::shared_ptr<IMemory> buf_mem_ew_b =
+      mover->GetBuffer(size_c * sizeof(DataType), elemwise->GetMemoryBank(1));
+  std::shared_ptr<IMemory> buf_mem_ew_c =
+      mover->GetBuffer(size_c * sizeof(DataType), elemwise->GetMemoryBank(2));
+
+  CYNQ_INFO("Loading input");
+  fill_data(buf_mem_mm_a, size_a, 1002);
+  fill_data(buf_mem_mm_b, size_b, 55);
+  fill_data(buf_mem_mm_c, size_c, 0, 0);
+  fill_data(buf_mem_ew_a, size_a, 1002);
+  fill_data(buf_mem_ew_b, size_b, 55);
+  fill_data(buf_mem_ew_c, size_c, 0, 0);
+
+  CYNQ_INFO("Configuring accelerators");
+#ifdef PROFILE_MODE
+  GET_PROFILE_INSTANCE(configuration_time, cynq_profiler);
+  configuration_time->reset();
+#endif
+  matmul->Write(XMATMUL_CONTROL_ADDR_A_ROWS_DATA, &a_rows, 1);
+  matmul->Write(XMATMUL_CONTROL_ADDR_B_COLS_DATA, &b_cols, 1);
+  matmul->Write(XMATMUL_CONTROL_ADDR_C_COLS_DATA, &c_cols, 1);
+  matmul->Attach(XMATMUL_CONTROL_ADDR_A_DATA, buf_mem_mm_a);
+  matmul->Attach(XMATMUL_CONTROL_ADDR_B_DATA, buf_mem_mm_b);
+  matmul->Attach(XMATMUL_CONTROL_ADDR_C_DATA, buf_mem_mm_c);
+
+  elemwise->Write(XELEMENTWISE_CONTROL_ADDR_SIZE_DATA, &size_c, 1);
+  elemwise->Write(XELEMENTWISE_CONTROL_ADDR_OP_DATA, &op, 1);
+  elemwise->Attach(XELEMENTWISE_CONTROL_ADDR_IN1_DATA, buf_mem_ew_a);
+  elemwise->Attach(XELEMENTWISE_CONTROL_ADDR_IN2_DATA, buf_mem_ew_b);
+  elemwise->Attach(XELEMENTWISE_CONTROL_ADDR_OUT_R_DATA, buf_mem_ew_c);
+#ifdef PROFILE_MODE
+  configuration_time->tick();
+#endif
+
+  CYNQ_INFO("Starting the Accelerator and Move Data");
+  CYNQ_INFO("Trigger Upload");
+#ifdef PROFILE_MODE
+  GET_PROFILE_INSTANCE(upload_time, cynq_profiler);
+  upload_time->reset();
+#endif
+  buf_mem_mm_a->Sync(SyncType::HostToDevice);
+  buf_mem_mm_b->Sync(SyncType::HostToDevice);
+  buf_mem_ew_a->Sync(SyncType::HostToDevice);
+  buf_mem_ew_b->Sync(SyncType::HostToDevice);
+#ifdef PROFILE_MODE
+  upload_time->tick();
+#endif
+
+  CYNQ_INFO("Starting Accelerators");
+#ifdef PROFILE_MODE
+  GET_PROFILE_INSTANCE(compute_time, cynq_profiler);
+  compute_time->reset();
+#endif
+  matmul->Start(StartMode::Once);
+  matmul->Sync();
+  elemwise->Start(StartMode::Once);
+  elemwise->Sync();
+#ifdef PROFILE_MODE
+  compute_time->tick();
+#endif
+
+  CYNQ_INFO("Trigger Download");
+#ifdef PROFILE_MODE
+  GET_PROFILE_INSTANCE(download_time, cynq_profiler);
+  download_time->reset();
+#endif
+  buf_mem_mm_c->Sync(SyncType::DeviceToHost);
+  buf_mem_ew_c->Sync(SyncType::DeviceToHost);
```

Comment:
from reading and the context I get the idea that this buffer for c represents an output? if this is the case maybe using another name for c might make sense, something like buff_mem_mm_output?

Reply:
I stick with this convention by this fact:

$$
C = A \cdot B 
$$


$$
C = A + B 
$$

It is because of a math notation